exchanges/exchange.go

```diff
@@ -1752,3 +1751,72 @@ func (b *Base) MatchSymbolCheckEnabled(symbol string, a asset.Item, hasDelimiter
 func (b *Base) IsPairEnabled(pair currency.Pair, a asset.Item) (bool, error) {
 	return b.CurrencyPairs.IsPairEnabled(pair, a)
 }
+
+// Bootstrap function allows for exchange authors to supplement or override common startup actions
+// If exchange.Bootstrap returns false or error it will not perform any other actions.
+// If it returns true, or is not implemented by the exchange, it will:
+// * Print debug startup information
+// * UpdateOrderExecutionLimits
+// * UpdateTradablePairs
+func Bootstrap(ctx context.Context, b IBotExchange) error {
+	if continueBootstrap, err := b.Bootstrap(ctx); !continueBootstrap || err != nil {
+		return err
+	}
+
+	if b.IsVerbose() {
+		wsURL := ""
+		wsEnabled := false
+		if w, err := b.GetWebsocket(); err != nil {
+			wsURL = w.GetWebsocketURL()
+			wsEnabled = w.IsEnabled()
+		}
+		log.Debugf(log.ExchangeSys, "%s Websocket: %s. (url: %s).\n", b.GetName(), common.IsEnabled(wsEnabled), wsURL)
+		b.PrintEnabledPairs()
+	}
+
+	errC := make(chan error, 16) // Blocking isn't a dealbreaker, but a little buffer is nice
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		a := b.GetAssetTypes(true)
+		for x := range a {
+			if aErr := b.UpdateOrderExecutionLimits(ctx, a[x]); aErr != nil && !errors.Is(aErr, common.ErrNotYetImplemented) {
+				errC <- fmt.Errorf("failed to set exchange order execution limits: %w", aErr)
+			}
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if b.GetEnabledFeatures().AutoPairUpdates {
+			if aErr := b.UpdateTradablePairs(ctx, false); aErr != nil {
```

Comment:
It was throwing a bunch of errors when update order execution limits was trying to match symbols with loaded pairs. This helps to minimise that issue. Total loading is down to ~9s which is pretty good. 
```diff
diff --git a/exchanges/exchange.go b/exchanges/exchange.go
index 5e906ce37..706af3811 100644
--- a/exchanges/exchange.go
+++ b/exchanges/exchange.go
@@ -1774,6 +1774,12 @@ func Bootstrap(ctx context.Context, b IBotExchange) error {
                b.PrintEnabledPairs()
        }

+       if b.GetEnabledFeatures().AutoPairUpdates {
+               if err := b.UpdateTradablePairs(ctx, false); err != nil {
+                       return err
+               }
+       }
+
        errC := make(chan error, 16) // Blocking isn't a dealbreaker, but a little buffer is nice
        var wg sync.WaitGroup
        wg.Add(1)
@@ -1787,15 +1793,6 @@ func Bootstrap(ctx context.Context, b IBotExchange) error {
                }
        }()

-       wg.Add(1)
-       go func() {
-               defer wg.Done()
-               if b.GetEnabledFeatures().AutoPairUpdates {
-                       if aErr := b.UpdateTradablePairs(ctx, false); aErr != nil {
-                               errC <- fmt.Errorf("failed to update tradable pairs: %w", aErr)
-                       }
-               }
-       }()
        wg.Wait()
        close(errC)
```

Reply:
Fixed 1763395e